enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -820,14 +820,8 @@ func (ff *BatchFileFetcher) bytestreamReadFiles(ctx context.Context, instanceNam
 		return fp0, nil
 	})
 	if err != nil {
-		// Ensure that an unavailable error is always returned so that
-		// the download can be retried if we hit a transient error.
-		if !status.IsUnavailableError(err) {
-			err = status.UnavailableError(err.Error())
-		}
```

Comment:
This code was working around the context cancellation issue in the x/sync package - if the original caller cancels their context then it would cause other actions to fail with a potentially non-retryable error. With the new package, this logic can be removed.

Reply:
Kinda slow still but double checking -- you're sure it's OK to remove this?

Reply:
yup, this was just working around the limitations of the old package. The singleflight package has unit tests that confirm this but I also sanity checked it here: https://go.dev/play/p/ksdZXjzd1sI